packages/protocol/contracts/Staking.sol

```diff
@@ -0,0 +1,687 @@
+     */
+    function depositAndLock(
+        address _subject,
+        uint256 _amount,
+        uint256 _lockPeriodInSec
+    )
+        external
+        onlyValidLockPeriod(_lockPeriodInSec)
+        returns (uint256 unlockTimeInSec_)
+    {
+        unlockTimeInSec_ = _depositAndLock(
+            _subject,
+            _amount,
+            _lockPeriodInSec,
+            msg.sender
+        );
+    }
+
+    /**
+     * External function to deposit and lock tokens for a beneficiary.
+     * @param _subject Subject address for which tokens are getting deposited.
+     * @param _amount amount of tokens getting deposited.
+     * @param _lockPeriodInSec lock period for the tokens.
+     * @param _beneficiary Address of the beneficiary for the lock.
+     * @return unlockTimeInSec_ Unlock times for the locks.
+     */
+    function depositAndLockFor(
+        address _subject,
+        uint256 _amount,
+        uint256 _lockPeriodInSec,
+        address _beneficiary
+    )
+        external
+        onlyValidLockPeriod(_lockPeriodInSec)
+        returns (uint256 unlockTimeInSec_)
+    {
+        unlockTimeInSec_ = _depositAndLock(
+            _subject,
+            _amount,
+            _lockPeriodInSec,
+            _beneficiary
+        );
+    }
+
+    /**
+     * External function to deposit and lock multiple tokens.
+     * @param _subjects Subject addresses for which tokens are getting deposited.
+     * @param _amounts Amounts of tokens getting deposited.
+     * @param _lockPeriodInSec Lock periods for the tokens.
+     * @return unlockTimeInSec_ Unlock times for the locks.
+     */
+    function depositAndLockMultiple(
+        address[] memory _subjects,
+        uint256[] memory _amounts,
+        uint256 _lockPeriodInSec
+    )
+        external
+        onlyValidLockPeriod(_lockPeriodInSec)
+        returns (uint256[] memory unlockTimeInSec_)
+    {
+        if (_subjects.length != _amounts.length) {
+            revert Staking_InvalidInputLength();
+        }
+
+        unlockTimeInSec_ = new uint256[](_subjects.length);
+
+        for (uint256 i = 0; i < _subjects.length; i++) {
+            unlockTimeInSec_[i] = _depositAndLock(
+                _subjects[i],
+                _amounts[i],
+                _lockPeriodInSec,
+                msg.sender
+            );
+        }
+    }
+
+    /**
+     * External function to deposit and lock multiple tokens.
+     * @param _subjects Subject addresses for which tokens are getting deposited.
+     * @param _amounts Amounts of tokens getting deposited.
+     * @param _lockPeriodInSec Lock periods for the tokens.
+     * @param _beneficiary Address of the beneficiary for the lock.
+     * @return unlockTimeInSec_ Unlock times for the locks.
+     */
+    function depositAndLockMultipleFor(
+        address[] memory _subjects,
+        uint256[] memory _amounts,
+        uint256 _lockPeriodInSec,
+        address _beneficiary
```

Comment:
similar to above, benf. can be array too?

Reply:
Keeping one for simplicity. Anyone can build wrapper.